### PR TITLE
Start Infra and add status 

### PR DESCRIPTION
**Pull request by wu-xm**

### CODE REVIEW

File: controllers/cluster/api/v1/cluster_types.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ type ClusterSpec struct {
 type ClusterStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+	CurrStatus string `json:"currStatus,omitempty"`
```
Comment (fanux) [inline controllers/cluster/api/v1/cluster_types.go:40]: CurrStatus-> Status

File: controllers/cluster/controllers/cluster_controller.go [hunk 2]
```diff
@@ -63,7 +63,13 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		logger.Debug("ignore not found cluster error: %v", err)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-
+	if cluster.Status.CurrStatus == "" {
+		cluster.Status.CurrStatus = "creating"
```
Comment (fanux) [inline controllers/cluster/controllers/cluster_controller.go:67]: Defining states using enums

File: controllers/cluster/controllers/cluster_controller.go [hunk 3]
```diff
@@ -82,15 +87,18 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
+		cluster.Spec.SSH.PkData = infra.Status.SSH.PkData
 	}
-
 	err = r.applier.ReconcileCluster(infra, hosts, cluster)
 	if err != nil {
 		logger.Error("reconcile cluster error: %v", err)
 		return ctrl.Result{}, err
 	}
-
+	cluster.Status.CurrStatus = "created"
```
Comment (fanux) [inline controllers/cluster/controllers/cluster_controller.go:97]: created->running

File: controllers/infra/api/v1/infra_types.go [hunk 4]
```diff
@@ -163,6 +163,7 @@ type InfraStatus struct {
 	SSH         v1bata1.SSH `json:"ssh,omitempty"`
 	Hosts       []Hosts     `json:"hosts,omitempty"`
 	Connections string      `json:"connections,omitempty"` // master0 ip:port
+	CurrStatus  string      `json:"currStatus,omitempty"`
```
Comment (fanux) [inline controllers/infra/api/v1/infra_types.go:166]: CurrStatus->Status

Comment (wu-xm) [reply controllers/infra/api/v1/infra_types.go:166]: ok

File: controllers/cluster/api/v1/cluster_types.go [hunk 5]
```diff
@@ -37,6 +37,27 @@ type ClusterSpec struct {
...
+	switch s {
+	case Unknown:
+		return "Unknown"
+	case Creating:
+		return "Creating"
+	case Running:
+		return "Running"
+	default:
+		return "Error"
+	}
```
Comment (zzjin) [inline controllers/cluster/api/v1/cluster_types.go:60]: Try Keep status name with k8s is better: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/

**Comment by fanux** [comment]
Why delete the test case?

**Review by zzjin — APPROVED** [review]

File: controllers/cluster/Makefile [hunk 6]
```diff
@@ -62,7 +62,8 @@ test: manifests generate fmt vet envtest ## Run tests.
 
 .PHONY: build
 build:  ## Build manager binary.
-	CGO_ENABLED=0 GOOS=linux go build -o bin/manager main.go
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/controller-cluster-amd64 main.go
```
Comment (cuisongliu) [inline controllers/cluster/Makefile:65]: There is something wrong with writing like this, which will affect the CI of the cluster image.

**Pull request merged**